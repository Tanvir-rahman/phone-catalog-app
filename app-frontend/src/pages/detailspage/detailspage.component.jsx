import React, { useEffect, Suspense } from 'react';
import { connect  } from 'react-redux';
import { withRouter } from 'react-router-dom';

import './detailspage.styles.scss';

import { fetchPhoneDetails } from '@store/phones/phone.actions';

import Spinner from '@components/spinner/spinner.component';
import PhoneDetailContainer from '@components/phone-detail/phone-detail.container';

export const DetailsPage = ({ fetchPhoneDetails, history, match }) => {
  useEffect(() => {
    fetchPhoneDetails(match.params.phoneId);
  }, [fetchPhoneDetails, match.params.phoneId]);

  return (
    <Suspense fallback={<Spinner />}>
      <button onClick={history.goBack}>
        Back
      </button>
      <PhoneDetailContainer />
    </Suspense>
  );
};


const mapDispatchToProps = dispatch => ({
  fetchPhoneDetails: phoneId => dispatch(fetchPhoneDetails(phoneId))
});

export default withRouter(connect(null, mapDispatchToProps)(DetailsPage));
