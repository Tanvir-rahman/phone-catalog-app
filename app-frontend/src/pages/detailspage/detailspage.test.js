import React from 'react';
import { shallow } from 'enzyme';
import { DetailsPage } from './detailspage.component';

describe('DetailsPage component', () => {
  let mockfetchPhoneDetails;
  let mockProps;
  let mockHistory;
  let wrapper;

  beforeEach(() => {
    mockfetchPhoneDetails= jest.fn();
    mockHistory = {
      goBack: jest.fn()
    };

    mockProps = {
      fetchPhoneDetails: mockfetchPhoneDetails,
      match: {
        params: {
          phoneId: 'iphone-11'
        }
      },
      history: mockHistory,
    };

    wrapper = shallow(<DetailsPage {...mockProps} />);
  });

  it('should render DetailsPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
