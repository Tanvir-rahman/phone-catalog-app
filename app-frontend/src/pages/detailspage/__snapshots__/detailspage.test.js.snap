// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component should render DetailsPage component 1`] = `
<Suspense
  fallback={<Spinner />}
  fetchPhoneDetails={[MockFunction]}
  history={
    Object {
      "goBack": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "phoneId": "iphone-11",
      },
    }
  }
>
  <button
    onClick={[MockFunction]}
  >
    Back
  </button>
  <Connect(Component) />
</Suspense>
`;
